fsck/fsck.c

```diff
@@ -707,6 +735,15 @@ static int check_name_dentry_set(struct exfat_de_iter *iter,
 			return -EINVAL;
 		}
 	}
+
+	if (BITMAP_GET(iter->name_hash_bitmap, hash)) {
```

Comment:
Exfat's name hash is enough to avoid hash collision? It would be meaningful to discuss about other hash algorithms such as sha128, although cpu usage is likely to increase.

Reply:
Switching to sha256, hash collisions are unavoidable too.

In my patches, if the hashes match, the filenames are compared.

Reply:
But sha128 or something can decrease the chance of hash collisions more than Exfat's name hash, and avoid the comparision.

Reply:
Yes, as you said, sha128 can greatly reduce hash collisions.

But if using sha128, bitmap cannot be used to detect whether there is the same hash. We need to allocate a lot of space to save sha128 hashes, and we need to compare them one by one, which takes up a lot of resources. I think the current implementation is a balanced use of memory, CPU, storage resources.

Reply:
sha128 also is 16-byte and recent CPU architectures supports acceleration for sha. I am not forcing you to change it but I am just saying let's discuss about it.

Reply:
the hash value of sha1 can be used as the bit position of bitmap as you implemented. but the bitmap size have to be 2^17(2^20 % 2^3).

Reply:
[> the hash value of sha1 can be used as the bit position of bitmap as you implemented. but the bitmap size have to be 2^17(2^20 % 2^3).]

Sorry, I am confused. As you said, much more memory is required.